File: app/src/main/java/eu/kanade/tachiyomi/ui/base/presenter/NucleusConductorLifecycleListener.java
package eu.kanade.tachiyomi.ui.base.presenter;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.view.View;

import com.bluelinelabs.conductor.Controller;

public class NucleusConductorLifecycleListener extends Controller.LifecycleListener {

    private static final String PRESENTER_STATE_KEY = "presenter_state";

    private NucleusConductorDelegate delegate;

    public NucleusConductorLifecycleListener(NucleusConductorDelegate delegate) {
        this.delegate = delegate;
    }

    @Override
    public void postCreateView(@NonNull Controller controller, @NonNull View view) {
        delegate.onTakeView(controller);
    }

    @Override
    public void preDestroyView(@NonNull Controller controller, @NonNull View view) {
        delegate.onDropView();
    }

    @Override
    public void preDestroy(@NonNull Controller controller) {
        delegate.onDestroy();
    }

    @Override
    public void onSaveInstanceState(@NonNull Controller controller, @NonNull Bundle outState) {
        outState.putBundle(PRESENTER_STATE_KEY, delegate.onSaveInstanceState());
    }

    @Override
    public void onRestoreInstanceState(@NonNull Controller controller, @NonNull Bundle savedInstanceState) {
        delegate.onRestoreInstanceState(savedInstanceState.getBundle(PRESENTER_STATE_KEY));
    }

}

File: app/src/main/java/eu/kanade/tachiyomi/ui/base/presenter/NucleusConductorLifecycleListener.kt
package eu.kanade.tachiyomi.ui.base.presenter

import android.os.Bundle
import android.view.View
import com.bluelinelabs.conductor.Controller

class NucleusConductorLifecycleListener(private val delegate: NucleusConductorDelegate<*>) : Controller.LifecycleListener() {

    override fun postCreateView(controller: Controller, view: View) {
        delegate.onTakeView(controller)
    }

    override fun preDestroyView(controller: Controller, view: View) {
        delegate.onDropView()
    }

    override fun preDestroy(controller: Controller) {
        delegate.onDestroy()
    }

    override fun onSaveInstanceState(controller: Controller, outState: Bundle) {
        outState.putBundle(PRESENTER_STATE_KEY, delegate.onSaveInstanceState())
    }

    override fun onRestoreInstanceState(controller: Controller, savedInstanceState: Bundle) {
        delegate.onRestoreInstanceState(savedInstanceState.getBundle(PRESENTER_STATE_KEY))
    }

    companion object {
        private const val PRESENTER_STATE_KEY = "presenter_state"
    }
}
